### PR TITLE
feat: add auth routes

### PR DESCRIPTION
**Pull request by BlazeEcran**

### CODE REVIEW

File: code/server/src/routers/auth-router.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import express from 'express';
+import { UserRepository } from '../repositories/user-repository.js';
+import { UserService } from '../services/user-service.js';
+import { UserController } from '../controllers/user-controller.js';
+
+const userRouter = express.Router();
```
Comment (sramstein) [inline code/server/src/routers/auth-router.js]: User router ???

File: code/server/src/controllers/user-controller.js [hunk 2]
```diff
@@ -47,13 +53,23 @@ export class UserController {
   createUser = async (req, res) => {
     try {
       const userData = req.body;
-
+
```
Comment (sramstein) [inline code/server/src/controllers/user-controller.js]: crypt the password please

File: code/server/src/controllers/user-controller.js [hunk 3]
```diff
@@ -47,13 +53,23 @@ export class UserController {
   createUser = async (req, res) => {
     try {
       const userData = req.body;
-
+      
       const user = await this.userService.createUser(userData);
 
-      res.status(201).json({ data: user });
+      res.status(201).json({ data: user, token: this.createToken(user.firstname, user.lastname) });
```
Comment (sramstein) [inline code/server/src/controllers/user-controller.js]: Use unique email

**Review by sramstein — CHANGES_REQUESTED** [review]

File: code/server/src/controllers/user-controller.js (hunk 3 above)
Comment (BlazeEcran) [reply code/server/src/controllers/user-controller.js]: pas daccord

File: code/server/src/routers/auth-router.js (hunk 1 above)
Comment (BlazeEcran) [reply code/server/src/routers/auth-router.js]: Done

File: code/server/src/controllers/user-controller.js (hunk 2 above)
Comment (BlazeEcran) [reply code/server/src/controllers/user-controller.js]: donce

**Review by Azurioh — APPROVED** [review]

**Pull request merged**